onnxruntime/core/framework/data_transfer.cc

```diff
@@ -9,6 +9,14 @@ common::Status IDataTransfer::CopyTensor(const Tensor& src, Tensor& dst) const {
   return CopyTensor(src, dst, 0);
 }
 
+common::Status IDataTransfer::CopyTensors(const Tensor* src, Tensor* dst, int size) const {
```

Comment:
IIRC the batch interface (CopyTensors) for copying tensors was needed for efficiency. This implementation is no different than calling CopyTensor 'size' times. I think the batch interface should be implemented by the individual data transfer implementations so that they can be optimized differently. The current mechanism doesn't provide any way to override.

Reply:
you're right. this is a default implementation. and sub-classes should override it. this avoids every data transfer class implementing one. Make sense?